### PR TITLE
add support for digitalocean_spaces_bucket

### PR DESCRIPTION
**Pull request by circa10a**
- Support for monitoring spaces buckets
- Update dependencies

Example metrics

```
# HELP digitalocean_spaces_bucket Spaces bucket and its details. Will always be 1 if exists
# TYPE digitalocean_spaces_bucket gauge
digitalocean_spaces_bucket{name="blahblahblahbucket",region="nyc3"} 1
# HELP digitalocean_spaces_bucket_created Spaces bucket's creation date in unix epoch format
# TYPE digitalocean_spaces_bucket_created gauge
digitalocean_spaces_bucket_creation_date{name="blahblahblahbucket",region="nyc3"} 1.643148934e+09
```

cc @metalmatze

### CODE REVIEW

**Comment by circa10a** [comment]
What are your thoughts on adding me as a collaborator? I really enjoy adding new features to this exporter 😁

File: main.go [hunk 1]
```diff
@@ -129,3 +142,8 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// spacesAccessKeysSet returns if the spaces access key id and the spaces access key secret are set
+func (c *Config) spacesAccessKeysSet() bool {
+	return c.SpacesAccessKeyID != "" && c.SpacesAccessKeySecret != ""
+}
```
Comment (metalmatze) [inline main.go:149]: I wouldn't hide this behind a helper function. You can just call this in the if twice.

**Comment by metalmatze** [comment]
This PR looks good! Thanks! :100: 

I would love it you could help maintain this. In fact, I'm not using DigitalOcean anymore and having someone with an actual need seems much more fitting. :+1:

**Comment by metalmatze** [comment]
I invited you as collaborator. Maybe you can reach out on Twitter, Slack or anywhere it's good for you to discuss some next steps. Pretty much everywhere I'm @metalmatze.

**Comment by circa10a** [comment]
> I invited you as collaborator. Maybe you can reach out on Twitter, Slack or anywhere it's good for you to discuss some next steps. Pretty much everywhere I'm @metalmatze.

Awesome! Are you on discord?

**Comment by metalmatze** [comment]
Yes! In fact we use Discord for Parca primarily. Check out the Discord link at the bottom on Parca.dev and should be able to find me there! 👋

File: main.go (hunk 1 above)
Comment (circa10a) [reply main.go]: Fixed 👍

File: collector/spaces.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+				level.Warn(c.logger).Log(
+					"msg", "can't list spaces buckets",
+					"err", err,
+				)
+				return
+			}
+
+			for _, bucket := range buckets {
+				labels := []string{
+					bucket.CreationDate.Format(time.RFC3339),
```
Comment (metalmatze) [inline collector/spaces.go:117]: This is probably not a good idea. Instead, we could have a separate metric where the creation timestamp is the value?

Comment (circa10a) [reply collector/spaces.go:117]: That's fair. What do you think about `float64(bucket.CreationDate.Unix())` ? @metalmatze

File: collector/spaces.go [hunk 3]
```diff
@@ -34,27 +35,31 @@ const spacesDomain = "%s.digitaloceanspaces.com"
...
 		accessKeySecret: accessKeySecret,
-
-		Active: prometheus.NewDesc(
+		Bucket: prometheus.NewDesc(
 			"digitalocean_spaces_bucket",
 			"Spaces bucket and its details. Will always be 1 if exists",
 			labels, nil,
 		),
+		BucketCreationDate: prometheus.NewDesc(
+			"digitalocean_spaces_bucket_creation_date",
```
Comment (metalmatze) [inline collector/spaces.go:52]: How about `digitalocean_spaces_bucket_created`. That would be on pair with what kube-state-metric does with creation timestamps.

**Comment by metalmatze** [comment]
One last comment, then we're ready to ship it! :blush: :rocket:

File: collector/spaces.go (hunk 3 above)
Comment (circa10a) [reply collector/spaces.go]: Pushed! @metalmatze

**Review by metalmatze — APPROVED** [review]
Amazing! Keep it coming! :blush:

**Pull request merged**